include/zephyr/device.h

```diff
@@ -62,13 +62,6 @@ extern "C" {
  */
 typedef int16_t device_handle_t;
 
-/*
- * The build assert will fail if device_handle_t changes size, which
- * means the alignment directives in the linker scripts and in
- * `gen_handles.py` must be updated.
- */
-BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
```

Comment:
Why did you remove this? This is about the size of `uint16_t` and not the alignment?

Reply:
OK, saw this comment now https://github.com/zephyrproject-rtos/zephyr/pull/51162#pullrequestreview-1137291177

Reply:
because we relied on size being 2 (hardcoded), like in the gen_handles.py script.